cpp/dolfinx/la/PETScOperator.cpp

```diff
@@ -21,10 +21,9 @@ PETScOperator::PETScOperator(Mat A, bool inc_ref_count) : _matA(A)
     PetscObjectReference((PetscObject)_matA);
 }
 //-----------------------------------------------------------------------------
-PETScOperator::PETScOperator(PETScOperator&& A) noexcept : _matA(nullptr)
+PETScOperator::PETScOperator(PETScOperator&& A) noexcept
```

Comment:
I don't think `std::exchange` is `noexcept`.

Reply:
Right, sorry about that. Interestingly enough, cppreference uses it in a `noexcept` function, despite not marking it `noexcept` [cf. https://en.cppreference.com/w/cpp/utility/exchange ].